### PR TITLE
Bulk gyro/acc reads, temperature sensor framework

### PR DESCRIPTION
**Pull request by digitalentity**
This PR changes gyro/acc drivers to read acceleration/temperature/gyro data in one transaction. This should save some bus overhead compared to reading acceleration/gyro separately.

Also introduced framework for temperature sensors. Currently only one sensor - gyroscope/MPU. This temperature readings could be shown in OSD or used in future to compensate for temperature drift of accelerometer.

### CODE REVIEW

File: src/main/fc/fc_tasks.c [hunk 1]
```diff
@@ -452,6 +460,13 @@ cfTask_t cfTasks[TASK_COUNT] = {
         .staticPriority = TASK_PRIORITY_MEDIUM,
     },
 
+    [TASK_TEMPERATURE] = {
+        .taskName = "TEMPERATURE",
+        .taskFunc = taskUpdateTemperature,
+        .desiredPeriod = TASK_PERIOD_HZ(10),      // 10 Hz
```
Comment (DzikuVx) [inline src/main/fc/fc_tasks.c]: I'm pretty sure it can go down to 1Hz and no one would notice

Comment (digitalentity) [reply src/main/fc/fc_tasks.c]: Yeah, probably. Temperature inertia is quite large.

**Comment by shellixyz** [comment]
Might also be used if no other alternative is available to compensate for changes in air density when reading airspeed with a Pitot tube.

**Comment by digitalentity** [comment]
@shellixyz actually temperature compensation of pitot is not needed. Aerodynamically we are interested in mass air flow across our control surfaces which is dependent on air density and TAS. The combined measurement is called IAS (indicated airspeed) and this is exaclty what's measured by pitot tube.

Stall speed is exactly the same at all altitudes/temperatures/air pressures if you look at IAS, however if you look at TAS - stall speed would be higher at high altitudes due to low air density.

From https://en.wikipedia.org/wiki/Indicated_airspeed:
>The IAS is an important value for the pilot because it is the indicated speeds which are specified in the aircraft flight manual for such important performance values as the stall speed. These speeds, in true airspeed terms, vary considerably depending upon density altitude. However, at typical civilian operating speeds, the aircraft's aerodynamic structure responds to dynamic pressure alone, and the aircraft will perform the same when at the same dynamic pressure. Since it is this same dynamic pressure that drives the airspeed indicator, an aircraft will always, for example, stall at the published indicated airspeed (for the current configuration) regardless of density, altitude or true airspeed.

**Comment by shellixyz** [comment]
Very interesting. I didn't think about that. Learning every day :smiley:. Still could be used for compensating the TAS if displayed in the OSD.

**Comment by digitalentity** [comment]
I think showing TAS in OSD could lead to confusion and complaints about stall happening at different airspeeds (TAS). Originally pitot driver was attempting to calculate TAS, but after I realized that IAS is more meaningful to the pilot (and by coincidence easier to calculate) - https://github.com/iNavFlight/inav/pull/1841/commits/1f70cb02a3cb0671a0d8f1257c781e2711fb4462 was merged to calculate IAS instead.

**Comment by DzikuVx** [comment]
Yes, IAS is all what is really required here

File: src/main/sensors/temperature.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+#pragma once
+
+#include "config/parameter_group.h"
+
+typedef enum tempSensor_e {
+    TEMP_GYRO = 0,
+    TEMP_COUNT
+} tempSensor_t;
```
Comment (junwoo091400) [inline src/main/sensors/temperature.h]: Wouldn't "tempSensor_e" be more 'enum-like' name? Like most other typedefs in codebase

**Comment by Pairan** [comment]
**Temperature**
just, as I'm not so deep in all the coding and the ideas, but would this thing be something simply to be connected to an UART or pad:

http://www.analog.com/media/en/technical-documentation/data-sheets/TMP35_36_37.pdf

A mate came up with this for minimosd lately ...

**Comment by fiam** [comment]
@Pairan acc/gyro, MCU and baro have their own builtin temperature sensors, so most pilots won’t need any additional ones. The one you linked might be handy to monitor temperature for an specific component (eg a camera inside the fuselage).

**Comment by Pairan** [comment]
ah, I see ... considering myself educated for the moment :)

**Pull request merged**